code/modules/projectiles/projectile.dm

```diff
@@ -62,15 +62,17 @@
 	return ..()
 
 /obj/item/projectile/proc/Range()
-    range--
-    if(damage && tile_dropoff)
-        damage = max(0, damage - tile_dropoff) // decrement projectile damage based on dropoff value for each tile it moves
-    if(stamina && tile_dropoff_s)
-        stamina = max(0, stamina - tile_dropoff_s) // as above, but with stamina
-    if(range <= 0 && loc)
-        on_range()
-    if(!damage && !stamina && !nodamage)
-        on_range()
+	range--
+	if(damage && tile_dropoff)
+		damage = max(0, damage - tile_dropoff) // decrement projectile damage based on dropoff value for each tile it moves
+	if(stamina && tile_dropoff_s)
+		stamina = max(0, stamina - tile_dropoff_s) // as above, but with stamina
+	if(range <= 0 && loc)
+		on_range()
+	if(!damage && !stamina)
+		if(tile_dropoff || tile_dropoff_s) // does the projectile have falloff to start with

```

Comment:
Can be put in the if statement above
if(!damage && !stamina && (tile_dropoff || tile_dropoff_s))

Reply:
Much better, added it in now.